lib/improver/tests/wind_direction/test_WindDirection.py

```diff
@@ -347,35 +355,76 @@ def test_returns_confidence(self):
 class Test_wind_dir_decider(IrisTest):
     """Test the wind_dir_decider function."""
 
-    def setUp(self):
-        """Initialise plugin and supply data for tests"""
-        self.plugin = WindDirection()
+    @ManageWarnings(
+        ignored_messages=["Casting complex values"],
+        warning_types=[np.ComplexWarning])
+    def test_runs_function_1st_member(self):
+        """First element has two angles directly opposite (90 & 270 degs).
+        Therefore the calculated mean angle of 180 degs is basically
+        meaningless with an r value of nearly zero. So the code substitutes the
+        wind direction taken from the first ensemble value in its place."""
+        cube = make_wdir_cube_222()
+        self.plugin = WindDirection(backup_method="first realization")
         self.plugin.wdir_complex = WIND_DIR_COMPLEX
         self.plugin.realization_axis = 0
         self.plugin.wdir_slice_mean = make_wdir_cube_222()[0]
         self.plugin.wdir_slice_mean.data = np.array([[180.0, 55.0],
                                                      [280.0, 0.0]])
         self.plugin.wdir_mean_complex = (
             self.plugin.deg_to_complex(self.plugin.wdir_slice_mean.data))
-        self.cube = make_wdir_cube_222()[0]
-
-    def test_runs_function(self):
-        """First element has two angles directly opposite (90 & 270 degs).
-        Therefore the calculated mean angle of 180 degs is basically
-        meaningless with an r value of nearly zero. So the code subistites the
-        wind direction taken from the first ensemble value in its place."""
-
         expected_out = np.array([[90.0, 55.0],
                                  [280.0, 0.0]])
         where_low_r = np.array([[True, False],
                                 [False, False]])
-
-        self.plugin.wind_dir_decider(where_low_r, self.cube.data)
+        self.plugin.wind_dir_decider(where_low_r, cube)
         result = self.plugin.wdir_slice_mean.data
 
         self.assertIsInstance(result, np.ndarray)
         self.assertArrayAlmostEqual(result, expected_out)
 
+    @ManageWarnings(
+        ignored_messages=["Casting complex values"],
+        warning_types=[np.ComplexWarning])
```

Comment:
Note for myself and other reviewers.  Contrary to appearances, this is OK (we are 99% sure...).  It's due to the treatment of masks in square neighbourhooding.  It may be a good idea to open a ticket to suppress this warning when it's a mask, or change the treatment of masks, to avoid this confusion.

Reply:
Review comment: can we use `warnings_list=True` argument please to check that this warning is ONLY thrown for the mask (ie precisely once each for this and the previous test).

Reply:
This isn't recognised by the ManageWarnings class and I think it is out of scope of this ticket to add this. I will open a follow-on ticket instead.

Reply:
Sorry, that's my bad, the syntax is slightly different.  What I'm envisaging is:
```
@ManageWarnings(
        ignored_messages=["Casting complex values"],
        warning_types=[np.ComplexWarning],
        record=True)
def test_for_whatever(self, warning_list=None):
      ...
      self.assertEqual(len(warning_list), 1)
```
More examples exist in the code eg: https://github.com/metoppv/improver/blob/master/lib/improver/tests/wind_gust_diagnostic/test_WindGustDiagnostic.py#L212

Reply:
On further discussion / reflection: yes, I think you're right, this is actually a follow on ticket to deal with the warning that is raised in neighbourhood processing.